internal/provider/group_data_source.go

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	netbird "github.com/netbirdio/netbird/management/client/rest"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ datasource.DataSource = &GroupDataSource{}
+
+func NewGroupDataSource() datasource.DataSource {
+	return &GroupDataSource{}
+}
+
+// GroupDataSource defines the data source implementation.
+type GroupDataSource struct {
+	client *netbird.Client
+}
+
+func (d *GroupDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_group"
+}
+
+func (d *GroupDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Read Group metadata and associated resources and peers, see [NetBird Docs](https://docs.netbird.io/how-to/manage-network-access#groups) for more information.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "Group ID",
+				Required:            true,
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: "Group name identifier",
+				Computed:            true,
+			},
+			"peers_count": schema.Int32Attribute{
```

Comment:
See [comment about *_count fields](https://github.com/netbirdio/terraform-provider-netbird/pull/1/files#r2081321210)

Reply:
Resolving in https://github.com/netbirdio/terraform-provider-netbird/pull/7